### PR TITLE
Go Vendoring Standardization

### PR DESCRIPTION
**Pull request by natebrennand**
This PR was mostly autogenerated.

Changes to expect:
- removal of `*_test.go` files in /vendor
- addition of non go files in /vendor
- addtions to README.md & Makefile

**Before your merge!**:

- Please carefully check the Makefile changes
- Ensure that the build is passing
- Check that drone is using the proper image (Clever/drone-go:1.5)

Feel free to ping @natebrennand or #golang for clarification or help.

If you'd like more of an explanation please join us in #golang.

### CODE REVIEW

File: vendor/github.com/Clever/kayvee-go/.drone.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+image: bradrydzewski/go:1.4
```
Comment (peggyl) [inline vendor/github.com/Clever/kayvee-go/.drone.yml]: Should we be vendoring a more recent version of kayvee if this one is on Go 1.4 and using Hipchat?

**Comment by peggyl** [comment]
@natebrennand Overall looks fine, but I thought only `.go` source files were supposed to go in the `vendor/` directory. I'm seeing CI files like `.drone.yml` and `.travis.yml`, Go test files, README and LICENSE files, etc.

**Comment by natebrennand** [comment]
@peggyl: idk why godep changed that behavior but it now copies in all files from imported packages & all child directories.

File: vendor/github.com/Clever/kayvee-go/.drone.yml (hunk 1 above)
Comment (natebrennand) [inline vendor/github.com/Clever/kayvee-go/.drone.yml]: Not much of a reason to until we upgrade to using kayvee-go/logger IMO

**Comment by peggyl** [comment]
LGTM otherwise

**Pull request merged**